src/mappings/nabla/creation.ts

```diff
@@ -0,0 +1,220 @@
+    })
+}
+
+export async function getSwapPoolsOfRouterForToken(
+    ctx: EventHandlerContext,
+    routerSs58Address: string,
+    tokenSs58Address: string
+) {
+    return await ctx.store.findOne(SwapPool, {
+        where: {
+            router: { id: routerSs58Address },
+            token: { id: tokenSs58Address },
+        },
+        relations: { token: true },
+    })
+}
+
+export async function createBackstopPool(
+    ctx: EventHandlerContext,
+    hexAddress: string
+) {
+    const ss58Address = hexToSs58(hexAddress)
+    if (await getBackstopPool(ctx, ss58Address)) {
+        // Unexpected error: backstop pool already exists at that address
+        return
+    }
+
+    const contract = new backstopPoolAbi.Contract(ctx, hexAddress)
+
+    const routerAddress = await contract.router()
+    const tokenAddress = await contract.asset()
+
+    if (routerAddress === ZERO_ADDRESS || tokenAddress === ZERO_ADDRESS) {
+        return undefined
+    }
+
+    const router = await getRouter(ctx, hexToSs58(routerAddress))
+    if (router === undefined) {
+        return undefined
+    }
+
+    const token = await getOrCreateNablaToken(ctx, tokenAddress)
+    const tokenContract = new erc20Abi.Contract(ctx, tokenAddress)
+
+    const backstop = new BackstopPool({
+        id: ss58Address,
+        name: await contract.name(),
+        symbol: await contract.symbol(),
+        lpTokenDecimals: await contract.decimals(),
+        router: router,
+        token: token,
+        reserves: await tokenContract.balanceOf(hexAddress),
+        totalSupply: await contract.totalSupply(),
+        paused: false,
+        apr: 0n,
+    })
+    await ctx.store.save(backstop)
+}
+
+export async function createRouter(
+    ctx: EventHandlerContext,
+    hexAddress: string
+) {
+    const ss58Address = hexToSs58(hexAddress)
+    if (await getRouter(ctx, ss58Address)) {
+        // Unexpected error: router already exists at that address
+        return
+    }
+
+    const router = new Router({
+        id: ss58Address,
+        paused: false,
+    })
+    await ctx.store.save(router)
+}
+
+export async function getOrCreateNablaToken(
+    ctx: EventHandlerContext,
+    hexAddress: string
+) {
+    const ss58Address = hexToSs58(hexAddress)
+    let nablaToken = await ctx.store.get(NablaToken, ss58Address)
+    if (!nablaToken) {
+        const contract = new erc20Abi.Contract(ctx, hexAddress)
+        nablaToken = new NablaToken({
+            id: ss58Address,
+            decimals: await contract.decimals(),
+            name: await contract.name(),
+            symbol: await contract.symbol(),
+        })
+        await ctx.store.save(nablaToken)
+    }
+    return nablaToken
+}
+
+export async function createSwapPool(
+    ctx: EventHandlerContext,
+    hexAddress: string
+) {
+    const ss58Address = hexToSs58(hexAddress)
+    if (await getSwapPool(ctx, ss58Address)) {
+        // Unexpected error: swap pool already exists at that address
```

Comment:
It would be certainly helpful to make this more visible.

Is this the right pattern to use `console.log()` in the indexer code? Would we be able to see that in the subsquid log views? It seems that there are no other `console.log()` in event processing code.

Reply:
Yes, you can use `console.log()` and we will see it, no problem. We are already using it for errors during processing e.g. [here](https://github.com/pendulum-chain/pendulum-squids/blob/993d2f9f2eed0ff21c319c03358b8af634eec5f2/src/processor.ts#L341).

Reply:
I added some log statements: https://github.com/pendulum-chain/pendulum-squids/pull/52/commits/8a543ff3b98523cb9c628724020f5f7568d9f57d

Reply:
Thanks 👍